geopandas/_vectorized.py

```diff
@@ -112,144 +60,47 @@ def from_shapely(data):
     # First try a fast path for pygeos if possible, but do this in a try-except
     # block because pygeos.from_shapely only handles Shapely objects, while
     # the rest of this function is more forgiving (also __geo_interface__).
-    if compat.USE_PYGEOS and compat.PYGEOS_SHAPELY_COMPAT:
-        if not isinstance(data, np.ndarray):
-            arr = np.empty(len(data), dtype=object)
-            with compat.ignore_shapely2_warnings():
-                arr[:] = data
-        else:
-            arr = data
-        try:
-            return pygeos.from_shapely(arr)
-        except TypeError:
-            pass
 
     out = []
 
     for geom in data:
-        if compat.USE_PYGEOS and isinstance(geom, pygeos.Geometry):
+        if isinstance(geom, BaseGeometry):
             out.append(geom)
-        elif isinstance(geom, BaseGeometry):
-            if compat.USE_PYGEOS:
-                out.append(_shapely_to_pygeos(geom))
-            else:
-                out.append(geom)
         elif hasattr(geom, "__geo_interface__"):
             geom = shapely.geometry.shape(geom)
-            if compat.USE_PYGEOS:
-                out.append(_shapely_to_pygeos(geom))
-            else:
-                out.append(geom)
+            out.append(geom)
         elif isna(geom):
             out.append(None)
         else:
             raise TypeError("Input must be valid geometry objects: {0}".format(geom))
 
-    if compat.USE_PYGEOS:
-        return np.array(out, dtype=object)
-    else:
-        # numpy can expand geometry collections into 2D arrays, use this
-        # two-step construction to avoid this
-        aout = np.empty(len(data), dtype=object)
-        with compat.ignore_shapely2_warnings():
-            aout[:] = out
-        return aout
+    return np.array(out, dtype=object)
```

Comment:
fixed

Reply:
actually, thinking about I guess that `np.array(out, dtype=object)` is correct here. The reason we did allocate the array and then assign was to avoid triggering the array interface of old shapely geometry objects. @jorisvandenbossche is that right?

Reply:
> actually, thinking about I guess that `np.array(out, dtype=object)` is correct here. The reason we did allocate the array and then assign was to avoid triggering the array interface of old shapely geometry objects. @jorisvandenbossche is that right?

Yes, that's correct. The geometry objects are now pure scalar objects without any array-like features (not iterable, no array interface, etc), so the pre-allocating and then assigning workaround shouldn't be needed anymore.